### ISSUE TITLE
UI gives confusing error messages

### ISSUE THREAD

**Issue by Dieterbe** [jaegertracing/jaeger#401]
when you hit search, but the backend request results in a non-200 ok such as a 504,
the UI says:

> There was an error querying for traces:
> Bad JSON returned from the Jaeger Query Service.

this is confusing and it should probably say "backend returned 504 - gateway timeout" or something

**Comment by golonzovsky** [comment]
same error with following logs from query service:
```
{"level":"info","ts":1508430188.7339435,"caller":"healthcheck/handler.go:46","msg":"Health Check server started","http-port":16687}
2017/10/19 16:23:08 gocql: unable to dial control conn 10.20.25.21: dial tcp 10.20.25.21:9042: getsockopt: connection refused
{"level":"fatal","ts":1508430188.7403657,"caller":"query/main.go:77","msg":"Failed to init storage builder","error":"gocql: unable to create session: control: unable to connect to initial hosts: dial tcp 10.20.25.21:9042: getsockopt: connection refused","stacktrace":"github.com/uber/jaeger/vendor/go.uber.org/zap.Stack\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/field.go:191\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).check\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:301\ngithub.com/uber/jaeger/vendor/go.uber.org/zap.(*Logger).Fatal\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/go.uber.org/zap/logger.go:235\nmain.main.func1\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/query/main.go:77\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:636\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:722\ngithub.com/uber/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/uber/jaeger/vendor/github.com/spf13/cobra/command.go:681\nmain.main\n\t/home/travis/gopath/src/github.com/uber/jaeger/cmd/query/main.go:117"}
```

**Comment by bboreham** [comment]
Same confusing UI error for me; via the browser console I could see the underlying error was:

```
{"data":null,"total":0,"limit":0,"offset":0,"errors":[{"code":500,"msg":"Could not find aggregation of distinct_services"}]}
```

because I had no data because my agent wasn't installed properly.

**Comment by yurishkuro** [comment]
@tiffon this looks like it could be a very simply fix.

Should we even be returning JSON from the server when the response is non-2xx? I don't know what the convention is in this case for REST APIs.

**Comment by tiffon** [comment]
@yurishkuro I think returning the error details in JSON for `>= 400` responses is fine, but right now the UI is not processing it correctly.

I've created jaegertracing/jaeger-ui#127 to address this.

**Comment by tiffon** [comment]
This was fixed by jaegertracing/jaeger-ui#133.

**Comment by yurishkuro** [comment]
👏
